cli/flox/src/commands/publish.rs

```diff
@@ -45,17 +53,26 @@ impl Publish {
         }
 
         let PublishTarget { target } = self.publish_target;
+        if self.cache.is_some() && self.key_file.is_none() {
+            bail!("--cache requires a signing key to be provided via --key-file.");
+        }
```

Comment:
instead of 

```
    #[bpaf(long)]
    cache: Option<Url>,

    #[bpaf(long)]
    key_file: Option<String>,
```

i think we want

```
struct Publish {
    #[bpaf(external(cache_args), optional)]
    cache: Option<CacheArgs>,
    ...
}

struct CacheArgs {
	#[bpaf(long("cache")]
    url: Url,
    #[bpaf(long)]
    key_file: PathBuf,
}
```


This will ensure that url and key_file only occur together.

Reply:
`CacheArgs` can also trivially be mapped into `NixCopyCache`